industrial_ci/src/env.sh

```diff
@@ -123,6 +123,10 @@ function set_ros_variables {
     "eloquent")
         ros2_defaults "bionic"
         ;;
+    "foxy")
+        ros2_defaults "bionic"
+        DEFAULT_DOCKER_IMAGE=
```

Comment:
Why not
```suggestion
        DEFAULT_DOCKER_IMAGE=osrf/ros2:nightly
```
This way, one can specify `DOCKER_IMAGE=osrf/ros2:nightly` (for rolling nightly version) or `ROS_DISTRO=foxy` (for next release only)

Reply:
Or `DOCKER_IMAGE=osrf/ros2:nightly` for now?

Reply:
> This way, one can specify DOCKER_IMAGE=osrf/ros2:nightly (for rolling nightly version) or ROS_DISTRO=foxy (for next release only)

I am not sure I understood what the intended result is from these 2 different configurations.
What are the 2 scenarios we are trying to cover with this change?
What would the `.travis.yml` line would look like for each?

> Or DOCKER_IMAGE=osrf/ros2:nightly for now?

This will work more consistently than `DEFAULT_DOCKER_IMAGE=osrf/ros2:nightly` as the `DEFAULT_DOCKER_IMAGE` gets overwritten for some `ROS_REPO` values (including the default one)

---

A few hidden assumptions / expectations from my part that came up:

- Clear at first glance
It should be clear to user which job is a nightly build vs a build against a released distribution
in ros2ci it was done by naming the job nightly
```yaml
    - env: JOB_TYPE=nightly
      script:
      - cp additional_repos.repos .ros2ci/
      - .ros2ci/travis.bash $JOB_TYPE
```
Not sure what would be the best for ICI. Maybe having the `DOCKER_IMAGE=osrf/ros2:nightly` at the beginning of the line is enough :man_shrugging: 
- Minimal effort to set up
Should be a one liner for user to set up in their CI system e.g.:
```diff
 env:
   matrix:
     - ROS_DISTRO="dashing" ROS_REPO=main
     - ROS_DISTRO="eloquent" ROS_REPO=main
+    - DOCKER_IMAGE="osrf/ros2:nightly" UPSTREAM_WORKSPACE="additional_repos.repos"
```

- Zero maintenance from users
Once added the line should never need to be updated to keep tracking "ROS 2 master"
  - line should be `ROS_DISTRO`-free
- Clearly separated from release jobs
The job defined will never track a released `ROS_DISTRO` from debians, if users want to add a job for a new distro they should
```diff
 env:
   matrix:
     - ROS_DISTRO="dashing" ROS_REPO=main
     - ROS_DISTRO="eloquent" ROS_REPO=main
+    - ROS_DISTRO="foxy" ROS_REPO=main
     - DOCKER_IMAGE="osrf/ros2:nightly" UPSTREAM_WORKSPACE="additional_repos.repos"
```

Reply:
> I am not sure I understood what the intended result is from these 2 different configurations.

1. test against night build (`DOCKER_IMAGE="osrf/ros2:nightly" UPSTREAM_WORKSPACE="additional_repos.repos`)
2. test against ROS2 foxy (`ROS_DISTRO="foxy" ROS_REPO=main`). Could point to the nightly build right now and can be switched to the testing/main repo as they become available.

Reply:
Thanks for clarifying :+1:

Yes it should work.
Users with all their dependencies in [ros2.repos](https://github.com/ros2/ros2/blame/master/ros2.repos) will be fine, the ones with extra dependencies will need to remove the `UPSTREAM_WORKSPACE` once the packages are available in the distro
```
ROS_DISTRO="foxy" ROS_REPO=main UPSTREAM_WORKSPACE="additional_repos.repos"
```
 to
```
ROS_DISTRO="foxy" ROS_REPO=main
```